### ISSUE TITLE
Error moving files during cluster synchronization

### ISSUE THREAD

**Issue by crd1985** [wazuh/wazuh#3669]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.3 | wazuh-clusterd | Manager | any | any |

The following error is thrown in some Docker environments when source and target file are located in different volumes:
```
2019/07/16 10:32:16 wazuh-clusterd: ERROR: [Worker wazuh-manager-worker-0] [Integrity] Error processing shared file '/etc/shared/default/merged.mg': [Errno 18] Invalid cross-device link: '/var/ossec/queue/cluster/wazuh-manager-worker-0/wazuh-manager-worker-0-1563273136.525216-07246990498296324.zipdir/etc/shared/default/merged.mg' -> '/var/ossec/etc/shared/default/merged.mg'
2019/07/16 10:32:16 wazuh-clusterd: ERROR: [Worker wazuh-manager-worker-0] [Integrity] Found errors: 1 overwriting, 0 creating and 0 removing
```

This is something similar to issue #3335 

The recommended fix is replacing `os.rename` by `shutil.move` using `copyfile` copy_function.

Care must be taken when target file does not have write permissions, otherwise a `Permission denied` error will arise.

**Comment by crd1985** [comment]
We need to ensure that the following files has write permissions, otherwise the cluster could fail synchronizing them:
- etc/client.keys
- etc/shared/merged.mg
- var/multigroups/merged.mg
- etc/rules/*
- etc/decoders/*
- etc/lists/*
- queue/agent-info/agent-info.merged
- queue/agent-groups/*

Furthermore, ossec.conf should have write permissions for ossec user to let the API edit it.

**Comment by Skeptor** [comment]
I'm currently working to change the permissions of files in the branch: https://github.com/wazuh/wazuh/tree/3669-fix-permissions

The commit https://github.com/wazuh/wazuh/commit/93eeceee28f7e35d3fca195a267c879f48ebfe45 change the installation script to create the files with write permissions for group `ossec`. 
It also changes the permissions of the `merged.md`

**Comment by crd1985** [comment]
Thanks @Skeptor . I have checked that now .cdb files have proper permissions (660).

I have combined all needed changes under branch https://github.com/wazuh/wazuh/tree/fix-3669-cluster-copyfile. 

I have also found an error in the framework that make a copy of `agent.conf` with bad permissions (644) when creating a new group.

I need to launch a cluster environment with all these fixes and test again. I hope this time all file permissions are OK.
